### DISCUSSION TITLE
Adding columns to distributed table

### DISCUSSION THREAD

**Discussion by adam-mrozik** [ClickHouse/ClickHouse#36788] [Q&A]
Hey, simple question from me:

I have table `my_table_local` and distributed table `my_table` pointing to `my_table_local` and I want to add an additional column.
Is the only way to do it adding column to local table, deleting Distributed table and recreating it (or alternatively adding columns to both local and distributed table)?

**Comment by den-crane** [comment]
Why is that? You can columns to Distributed tables. Just execute the same `alter add/modify/drop column` commands.

**Comment by alexey-milovidov** [comment]
Yes, and it will work perfectly even if you have different structure in Distributed and in the target tables.

**Comment by adam-mrozik** [comment]
I found it to not be necessarily true. My workflow goes as follows:

```
Create table my_table_local ON CLUSTER '{cluster}'
...
Create Table my_table ON CLUSTER '{cluster}'  as my_table_local Engine=Distributed(cluster, database, my_table_local, rand());

ALTER TABLE my_table ADD COLUMN xyz


INSERT INTO my_table ....

ClickhouseError: my_table_local has no column xyz
```

**Reply by rajatkooapp** [reply]
Alter both the tables.
`ALTER TABLE my_table_local on cluster <your-cluster-name> ADD COLUMN xyz`
`ALTER TABLE my_table on cluster <your-cluster-name> ADD COLUMN xyz`

**Reply by adam-mrozik** [reply]
@rajatkooapp yeah, this is what I do right now. Was just wondering if I can avoid always doing that, since Distributed was supposed to be just a view pointing to this table

**Reply by rajatkooapp** [reply]
don't think so, as @alexey-milovidov mentioned, that it can sometimes be the case where you want different columns than your original table columns. makes sense to alter both the tables separately.
